### PR TITLE
UOE-10590: Do not start service if failed to load profileMetaData from DB

### PR DESCRIPTION
**Pull request by AvinashKapre**

### CODE REVIEW

File: modules/pubmatic/openwrap/openwrap.go [hunk 1]
```diff
@@ -95,7 +95,12 @@ func initOpenWrap(rawCfg json.RawMessage, moduleDeps moduledeps.ModuleDeps) (Ope
 		Cache:                 owCache,
 		ProfileMetaDataExpiry: cfg.Cache.ProfileMetaDataCacheExpiry,
 	})
-	profileMetaData.Start()
+	if err = profileMetaData.Start(); err != nil {
+		glog.Error("Failed to load profileMetaData from DB")
+		return OpenWrap{}, fmt.Errorf("error while initializing profile-metadata: %v", err)
+	} else {
```
Comment (ashishshinde-pubm) [inline modules/pubmatic/openwrap/openwrap.go:101]: no need to write else statement

Comment (pm-saurabh-narkhede) [inline modules/pubmatic/openwrap/openwrap.go:101]: Do we need else block if we are returning from `if`?

File: modules/pubmatic/openwrap/profilemetadata/reloader.go [hunk 2]
```diff
@@ -53,14 +60,25 @@ func (pmd *profileMetaData) Stop() {
...
 	ticker := time.NewTicker(time.Duration(pmd.profileMetaDataExpiry) * time.Second)
 	for {
-		//Populating pmdature config maps from cache
-		pmd.updateProfileMetaDataMaps()
+		//Populating pmdata config maps from cache (if data is not loaded from DB for first instance then do not start the service)
+		if err := pmd.updateProfileMetaDataMaps(); err != nil {
+			if firstdbLoad {
+				pmd.failToLoadDBData <- true
+				return
+			} else {
```
Comment (pm-saurabh-narkhede) [inline modules/pubmatic/openwrap/profilemetadata/reloader.go]: here also no need of else block

**Review by pm-saurabh-narkhede — APPROVED** [review]

**Review by pm-nilesh-chate — APPROVED** [review]

File: modules/pubmatic/openwrap/profilemetadata/reloader.go [hunk 3]
```diff
@@ -53,14 +60,24 @@ func (pmd *profileMetaData) Stop() {
...
+		//Populating pmdata config maps from cache (if data is not loaded from DB for first instance then do not start the service)
+		if err := pmd.updateProfileMetaDataMaps(); err != nil {
+			if firstdbLoad {
+				pmd.failToLoadDBData <- true
+				return
+			}
+			glog.Error("Failed to load profileMetaData from DB", err.Error())
+		} else {
+			firstdbLoad = false
+			pmd.failToLoadDBData <- false
```
Comment (Pubmatic-Dhruv-Sonone) [inline modules/pubmatic/openwrap/profilemetadata/reloader.go:79]: `pmd.failToLoadDBData <- false ` will block on second run,  as value will not be read from the channel.  Send the value on the **pmd.failToLoadDBData** only on the first run.

**Pull request merged**